### PR TITLE
GODRIVER-1582 Add support for commitQuorum

### PR DESCRIPTION
**Pull request by divjotarora**

### CODE REVIEW

File: design.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+- Add CommitQuorum to CreateIndexesOptions
```
Comment (jyemin) [inline design.md:1]: I'm guessing you didn't mean to add this file to the PR.

File: mongo/options/indexoptions.go [hunk 2]
```diff
@@ -13,6 +13,20 @@ import (
 // CreateIndexesOptions represents options that can be used to configure IndexView.CreateOne and IndexView.CreateMany
 // operations.
 type CreateIndexesOptions struct {
+	// The number of data-bearing members of a replica set, including the primary, that must complete the index builds
+	// successfully before the primary marks the indexes as ready. This should either be a string or int32 value. The
+	// semantics of the values are as follows:
+	//
+	// 1. String: specifies a tag. All members with that tag must complete the build.
+	// 2. int: the number of memebers that must complete the build.
```
Comment (jyemin) [inline mongo/options/indexoptions.go:21]: memebers => members

File: mongo/options/indexoptions.go [hunk 3]
```diff
@@ -13,6 +13,20 @@ import (
...
+	// 3. "majority": A special value to indicate that more than half the nodes must complete the build.
+	// 4. "voitingMembers": A special value to indicate that all voting data-bearing nodes must complete.
```
Comment (jyemin) [inline mongo/options/indexoptions.go:23]: voitingMembers => votingMembers

File: design.md (hunk 1 above)
Comment (divjotarora) [reply design.md:1]: Whoops. Removed.

File: mongo/options/indexoptions.go (hunk 2 above)
Comment (divjotarora) [reply mongo/options/indexoptions.go:21]: Done.

File: mongo/options/indexoptions.go (hunk 3 above)
Comment (divjotarora) [reply mongo/options/indexoptions.go:23]: Done.

**Pull request merged**